### PR TITLE
Sync update with neo pull 2119

### PR DESCRIPTION
**Pull request by chenzhitong**
Remove `Contract.Create`, `Contract.Update`, `Contract.Destroy`;
~~Add `Contract.CallNative`.~~

### CODE REVIEW

**Comment by chenzhitong** [comment]
Need to test in conjunction with https://github.com/neo-project/neo/pull/2119/

File: src/Neo.SmartContract.Framework/Services/Neo/Contract.cs [hunk 1]
```diff
@@ -28,14 +28,8 @@ public class Contract
...
-        [Syscall("System.Contract.Create")]
-        public static extern Contract Create(byte[] script, string manifest);
-
-        [Syscall("System.Contract.Update")]
-        public static extern void Update(byte[] script, string manifest);
-
-        [Syscall("System.Contract.Destroy")]
-        public static extern void Destroy();
+        [Syscall("System.Contract.CallNative")]
+        public static extern object CallNative(string name);
```
Comment (ProDog) [inline src/Neo.SmartContract.Framework/Services/Neo/Contract.cs:32]: How to call it in the contract?😅

Comment (Tommo-L) [reply src/Neo.SmartContract.Framework/Services/Neo/Contract.cs:32]: Remove this method, it's not for normal contract.

File: src/Neo.SmartContract.Framework/Services/Neo/Contract.cs [hunk 2]
```diff
@@ -28,14 +28,8 @@ public class Contract
         [Syscall("System.Contract.CallEx")]
         public static extern object CallEx(UInt160 scriptHash, string method, object[] arguments, CallFlags flag);
 
-        [Syscall("System.Contract.Create")]
-        public static extern Contract Create(byte[] script, string manifest);
-
-        [Syscall("System.Contract.Update")]
-        public static extern void Update(byte[] script, string manifest);
```
Comment (Tommo-L) [inline src/Neo.SmartContract.Framework/Services/Neo/Contract.cs:35]: Add `ManagementContract` contract.

File: src/Neo.SmartContract.Framework/Services/Neo/Contract.cs (hunk 1 above)
Comment (ProDog) [reply src/Neo.SmartContract.Framework/Services/Neo/Contract.cs:32]: Need to support `Update` and `Destroy`.

**Comment by ProDog** [comment]
According to neo-project/neo#2119, new native contract hash is:
```
NEO hash: 0x74c21a1ca66b7a190bf2a65db83ba6fe550cea64
GAS hash: 0xb399c051778cf37a1e4ef88509b2e054d0420a32
ManagementContract hash: 0xdf18cb2476964c241558ed1e2e8881dcd2d50bde
```

File: src/Neo.SmartContract.Framework/Services/Neo/GAS.cs [hunk 3]
```diff
@@ -9,6 +9,8 @@ public class GAS
     {
         public static extern UInt160 Hash { [ContractHash] get; }
         public static extern string Name { get; }
+        public static extern int Id { get; }
+        public static extern uint ActiveBlockIndex { get; }
```
Comment (Tommo-L) [inline src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:13]: Remove this two methods, which weren't defined for contract.

Comment (erikzhang) [reply src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:13]: If you remove these fields, the fifth field will become the third one, which will be incorrect.

File: templates/Template.CSharp/Contract1.cs [hunk 4]
```diff
@@ -7,44 +7,44 @@
...
+        if (update) return;
+
+        // It will be executed during deploy
+        Storage.Put("Hello", "World");
     }
+
+    public static void Update(byte[] nefFile, string manifest)
+    {
+        if (!IsOwner()) throw new Exception("No authorization.");
+        Contract.Call(ManagementContract.Hash, "update", nefFile, manifest);
```
Comment (superboyiii) [inline templates/Template.CSharp/Contract1.cs:41]:
```suggestion
        Contract.Call(ManagementContract.Hash, "update", new object[] { nefFile, manifest });
```

File: templates/Template.CSharp/Contract1.cs [hunk 5]
```diff
@@ -7,44 +7,44 @@
...
+    }
+
+    public static void Destroy()
+    {
+        if (!IsOwner()) throw new Exception("No authorization.");
+        Contract.Call(ManagementContract.Hash, "destroy");
```
Comment (superboyiii) [inline templates/Template.CSharp/Contract1.cs:47]:
```suggestion
        Contract.Call(ManagementContract.Hash, "destroy", new object[] {});
```

File: tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_CreateAndUpdate.cs [hunk 6]
```diff
@@ -4,9 +4,10 @@ namespace Neo.Compiler.MSIL.TestClasses
 {
     public class Contract_CreateAndUpdate : SmartContract.Framework.SmartContract
     {
-        public static int OldContract(byte[] script, string manifest)
+        public static int OldContract(byte[] nefFile, string manifest)
         {
             Contract.Update(script, manifest);
+            Contract.Call(ManagementContract.Hash, "update", nefFile, manifest);
```
Comment (superboyiii) [inline tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_CreateAndUpdate.cs:10]:
```suggestion
            Contract.Call(ManagementContract.Hash, "update", new object[] { nefFile, manifest });
```

**Review by superboyiii — CHANGES_REQUESTED** [review]

**Comment by shargon** [comment]
Wait for https://github.com/neo-project/neo-devpack-dotnet/pull/391

File: src/Neo.SmartContract.Framework/Services/Neo/ManagementContract.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Neo.SmartContract.Framework.Services.Neo
+{
+    [Contract("0xdf18cb2476964c241558ed1e2e8881dcd2d50bde")]
+    class ManagementContract
```
Comment (cloud8little) [inline src/Neo.SmartContract.Framework/Services/Neo/ManagementContract.cs:8]: should be public

File: src/Neo.SmartContract.Framework/Services/Neo/GAS.cs (hunk 3 above)
Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:12]: It can be private

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:12]: But ActiveBlockIndex was no exported

Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:12]: We can use `private`

**Comment by shargon** [comment]
Merge?

**Review by shargon — DISMISSED** [review]

**Comment by superboyiii** [comment]
I found native hash was changed again in the latest neo master.
```
NEO hash: 0x0a46e2e37c9987f570b4af253fb77e7eef0f72b6
GAS hash: 0xa6a6c15dcdc9b997dac448b6926522d22efeedfb
```
Do you think it's good to be fixed in this PR or create another? @chenzhitong @shargon

**Review by shargon — DISMISSED** [review]

File: src/Neo.SmartContract.Framework/Services/Neo/GAS.cs (hunk 3 above)
Comment (shargon) [reply src/Neo.SmartContract.Framework/Services/Neo/GAS.cs:12]: I tested without `ActiveBlockIndex` and it works well, I think that we should remove it

**Review by shargon — DISMISSED** [review]

**Comment by superboyiii** [comment]
Merge?

**Review by shargon — DISMISSED** [review]

**Comment by superboyiii** [comment]
@erikzhang Could you review this?

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**